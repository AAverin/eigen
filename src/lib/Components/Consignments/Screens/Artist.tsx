import * as React from "react"

import ArtistSearch from "../Components/ArtistSearchResults"
import DoneButton from "../Components/BottomAlignedButton"
import ConsignmentBG from "../Components/ConsignmentBG"

import { ArtistResult, ConsignmentSetup } from "../index"

import { debounce } from "lodash"
import { NavigatorIOS, Route, ScrollView, View, ViewProperties } from "react-native"
import metaphysics from "../../../metaphysics"

interface ArtistSearchResponse {
  match_artist: ArtistResult[]
}

interface Props extends ConsignmentSetup {
  navigator: NavigatorIOS
  route: Route
  updateWithResult?: (result: ArtistResult) => void
}

interface State {
  query: string
  searching: boolean
  results: any | null
}

export default class Artist extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      query: null,
      searching: false,
      results: null,
    }
  }

  doneTapped = () => {
    this.props.navigator.pop()
  }

  artistSelected = (result: ArtistResult) => {
    this.props.navigator.pop()
    this.props.updateWithResult(result)
  }

  textChanged = async (text: string) => {
    this.setState({ query: text, searching: text.length > 0 })
    this.searchForQuery(text)
  }

  searchForQuery = async (query: string) => {
    const results = await metaphysics<ArtistSearchResponse>(`
      {
        match_artist(term: "${query}") {
          id
          name
          image {
            url
          }
        }
      }
    `)
    this.setState({ results: results.match_artist, searching: false })
  }

  render() {
    // This might become a higher order component for reuse, if used more elsewhere
    const doneButtonStyles = {
      backgroundColor: "black",
      marginBottom: 20,
      paddingTop: 18,
      height: 56,
    }

    return (
      <ConsignmentBG>
        <DoneButton onPress={this.doneTapped} bodyStyle={doneButtonStyles} buttonText="DONE">
          <View
            style={{ alignContent: "center", justifyContent: "flex-end", flexGrow: 1, marginLeft: 20, marginRight: 20 }}
          >
            <ArtistSearch
              results={this.state.results}
              query={this.state.query}
              onChangeText={this.textChanged}
              searching={this.state.searching}
              resultSelected={this.artistSelected}
            />
          </View>
        </DoneButton>
      </ConsignmentBG>
    )
  }
}
