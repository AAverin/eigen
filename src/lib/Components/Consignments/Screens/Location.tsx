import * as React from "react"

import Search from "../Components/ArtistSearchResults"
import ConsignmentBG from "../Components/ConsignmentBG"
import DoneButton from "../Components/DoneButton"

import { ConsignmentSetup, SearchResult } from "../index"

import { debounce } from "lodash"
import { NavigatorIOS, Route, ScrollView, View, ViewProperties } from "react-native"
import metaphysics from "../../../metaphysics"

import { stringify } from "qs"

import { NativeModules } from "react-native"
const { Emission } = NativeModules

interface ArtistSearchResponse {
  match_artist: SearchResult[]
}

interface Props extends ConsignmentSetup, ViewProperties {
  navigator: NavigatorIOS
  route: Route
  updateWithResult?: (result: SearchResult) => void
}

interface State {
  query: string
  searching: boolean
  results: any | null
}

export default class Location extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      query: null,
      searching: false,
      results: null,
    }
  }

  doneTapped = () => {
    this.props.navigator.pop()
  }

  locationSelected = (result: SearchResult) => {
    this.props.navigator.pop()
    this.props.updateWithResult(result)
  }

  textChanged = async (text: string) => {
    this.setState({ query: text, searching: text.length > 0 })
    this.searchForQuery(text)
  }

  searchForQuery = async (query: string) => {
    // The 2nd is a throw-away key on a throw-away account
    const apiKey = Emission.googleMapsAPIKey || "AIzaSyBJRIy_zCXQ7XYt9Ubn8bpUIEAxEOKUmx8"
    const queryString = stringify({
      key: apiKey,
      language: "en",
      types: "(cities)",
      input: query,
    })

    const response = await fetch("https://maps.googleapis.com/maps/api/place/autocomplete/json?" + queryString)
    const results = await response.json()

    this.setState({
      results: results.predictions && results.predictions.map(this.predictionToResult),
      searching: false,
    })
  }

  predictionToResult = prediction => ({
    id: prediction.place_id,
    name: prediction.description,
  })

  render() {
    return (
      <ConsignmentBG>
        <DoneButton onPress={this.doneTapped}>
          <View
            style={{ alignContent: "center", justifyContent: "flex-end", flexGrow: 1, marginLeft: 20, marginRight: 20 }}
          >
            <Search
              results={this.state.results}
              query={this.state.query}
              onChangeText={this.textChanged}
              searching={this.state.searching}
              resultSelected={this.locationSelected}
              placeholder="Artist/Designer Name"
              noResultsMessage="Unfortunately we are not accepting consignments for works by"
            />
          </View>
        </DoneButton>
      </ConsignmentBG>
    )
  }
}
