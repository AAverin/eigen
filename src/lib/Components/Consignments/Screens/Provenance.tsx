import * as React from "react"

import ConsignmentBG from "../Components/ConsignmentBG"
import DoneButton from "../Components/DoneButton"

import { NavigatorIOS, Route, View } from "react-native"
import TextArea, { TextAreaProps } from "../Components/TextArea"
import { ConsignmentSetup } from "../index"

interface Props extends ConsignmentSetup {
  navigator: NavigatorIOS
  route: Route
  updateWithProvenance?: (provenance: string) => void
}

interface State {
  provenance: string
}

export default class Provenance extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      provenance: null,
    }
  }

  doneTapped = () => {
    this.props.navigator.pop()
  }

  render() {
    return (
      <ConsignmentBG>
        <DoneButton onPress={this.doneTapped}>
          <View
            style={{ alignContent: "center", justifyContent: "flex-end", flexGrow: 1, marginLeft: 20, marginRight: 20 }}
          >
            <TextArea
              text={{
                placeholder:
                  "Add notes about how you aquired the work. If you’re not sure add any details about how long you’ve had the work.",
                autoFocus: typeof jest === "undefined" /* TODO: https://github.com/facebook/jest/issues/3707 */,
              }}
            />
          </View>
        </DoneButton>
      </ConsignmentBG>
    )
  }
}
