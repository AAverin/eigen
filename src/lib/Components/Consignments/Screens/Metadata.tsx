import * as React from "react"

import {
  Keyboard,
  LayoutAnimation,
  NativeMethodsMixinStatic,
  NavigatorIOS,
  Picker,
  Route,
  ScrollView,
  TextInput,
  TouchableHighlight,
  TouchableWithoutFeedback,
  View,
  ViewProperties,
} from "react-native"
import { ConsignmentMetadata, SearchResult } from "../"
import { Fonts } from "../../../../data/fonts"
import TODO from "../Components/ArtworkConsignmentTodo"
import ConsignmentBG from "../Components/ConsignmentBG"
import DoneButton from "../Components/DoneButton"
import { Form, Label, Row } from "../Components/FormElements"
import Text from "../Components/TextInput"
import Toggle from "../Components/Toggle"
import { BodyText, LargeHeadline, Subtitle } from "../Typography"
import Artist from "./Artist"
import Provenance from "./Provenance"
import SelectFromPhotoLibrary from "./SelectFromPhotoLibrary"
import Welcome from "./Welcome"

interface Props extends ViewProperties {
  navigator: NavigatorIOS
  route: Route // this gets set by NavigatorIOS
  updateWithMetadata?: (result: ConsignmentMetadata) => void
  metadata: ConsignmentMetadata
}

// See: https://github.com/artsy/force/blob/814a03a579290eaac74f910a0db28c2afd9b1753/desktop/apps/consign/client/reducers.js#L22-L38
const categoryOptions = [
  { name: "Painting", value: "PAINTING" },
  { name: "Sculpture", value: "SCULPTURE" },
  { name: "Photography", value: "PHOTOGRAPHY" },
  { name: "Print", value: "PRINT" },
  { name: "Drawing, Collage or other Work on Paper", value: "DRAWING_COLLAGE_OR_OTHER_WORK_ON_PAPER" },
  { name: "Mixed Media", value: "MIXED_MEDIA" },
  { name: "Performance Art", value: "PERFORMANCE_ART" },
  { name: "Installation", value: "INSTALLATION" },
  { name: "Video/Film/Animation", value: "VIDEO_FILM_ANIMATION" },
  { name: "Architecture", value: "ARCHITECTURE" },
  { name: "Fashion Design and Wearable Art", value: "FASHION_DESIGN_AND_WEARABLE_ART" },
  { name: "Jewelry", value: "JEWELRY" },
  { name: "Design/Decorative Art", value: "DESIGN_DECORATIVE_ART" },
  { name: "Textile Arts", value: "TEXTILE_ARTS" },
  { name: "Other", value: "OTHER" },
]

interface State extends ConsignmentMetadata {
  showPicker?: boolean
}

// TODO: write a blog post about the refs

// This is the interface for what a component looks like when it's been set via `ref`.
// We do this so that pressing return can loop through text inputs and move on to the
// next empty one.
interface LiveStyledTextInput {
  /** The object which styled components wraps */
  root?: {
    /** A focus function for the text input */
    focus?: () => void
    props?: {
      /** The text value */
      value?: string
    }
  }
}

export default class Metadata extends React.Component<Props, State> {
  private yearInput: LiveStyledTextInput
  private mediumInput: LiveStyledTextInput
  private widthInput: LiveStyledTextInput
  private heightInput: LiveStyledTextInput
  private depthInput: LiveStyledTextInput

  constructor(props) {
    super(props)
    this.state = props.metadata || {}
  }

  doneTapped = () => {
    this.props.updateWithMetadata(this.state)
    this.props.navigator.pop()
  }

  updateUnit = () => this.setState({ unit: this.state.unit === "CM" ? "IN" : "CM" })
  updateTitle = title => this.setState({ title })
  updateYear = year => this.setState({ year })
  updateMedium = medium => this.setState({ medium })
  updateWidth = width => this.setState({ width })
  updateHeight = height => this.setState({ height })
  updateDepth = depth => this.setState({ depth })

  animateStateChange = newState => {
    const animate = LayoutAnimation.easeInEaseOut as any
    animate()
    this.setState(newState)
  }

  showCategorySelection = () => {
    Keyboard.dismiss()
    this.animateStateChange({ showSelector: true })
  }
  hideCategorySelection = () => this.animateStateChange({ showSelector: false })
  changeCategoryValue = (value, index) => {
    this.setState({
      categoryName: categoryOptions[index].name,
      category: categoryOptions[index].value,
    })
  }

  selectNextInput = () => {
    const inputs = [this.yearInput, this.mediumInput, this.widthInput, this.heightInput, this.depthInput]
    for (const input of inputs) {
      if (input && input.root && input.root.focus && !input.root.props.value) {
        return input.root.focus()
      }
    }
  }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <ConsignmentBG>
          <DoneButton onPress={this.doneTapped}>
            <ScrollView keyboardShouldPersistTaps="handled">
              <View style={{ padding: 10 }}>
                <Row>
                  <Text
                    text={{
                      placeholder: "Title",
                      onFocus: this.hideCategorySelection,
                      onChangeText: this.updateTitle,
                      value: this.state.title,
                      onSubmitEditing: this.selectNextInput,
                    }}
                    style={{ margin: 10 }}
                  />
                </Row>

                <Row>
                  <Text
                    text={{
                      placeholder: "Year",
                      onChangeText: this.updateYear,
                      value: this.state.year,
                      onFocus: this.hideCategorySelection,
                      onSubmitEditing: this.selectNextInput,
                      ref: component => (this.yearInput = component),
                    }}
                    style={{ margin: 10 }}
                  />
                </Row>

                <TouchableWithoutFeedback onPress={this.showCategorySelection}>
                  <Row>
                    <Text
                      text={{
                        placeholder: "Category",
                        value: this.state.categoryName,
                      }}
                      readonly={true}
                      style={{ margin: 10 }}
                    />
                  </Row>
                </TouchableWithoutFeedback>

                <Row>
                  <Text
                    text={{
                      placeholder: "Medium",
                      onChangeText: this.updateMedium,
                      value: this.state.medium,
                      onFocus: this.hideCategorySelection,
                      onSubmitEditing: this.selectNextInput,
                      ref: component => (this.mediumInput = component),
                    }}
                    style={{ margin: 10 }}
                  />
                </Row>

                <Row>
                  <Text
                    text={{
                      placeholder: "Width",
                      onChangeText: this.updateWidth,
                      value: this.state.width,
                      onFocus: this.hideCategorySelection,
                      onSubmitEditing: this.selectNextInput,
                      ref: component => (this.widthInput = component),
                    }}
                    style={{ margin: 10 }}
                  />
                  <Text
                    text={{
                      placeholder: "Height",
                      onChangeText: this.updateHeight,
                      value: this.state.height,
                      onFocus: this.hideCategorySelection,
                      onSubmitEditing: this.selectNextInput,
                      ref: component => (this.heightInput = component),
                    }}
                    style={{ margin: 10 }}
                  />
                </Row>

                <Row>
                  <Text
                    text={{
                      keyboardType: "numeric",
                      placeholder: "Depth",
                      onChangeText: this.updateDepth,
                      onFocus: this.hideCategorySelection,
                      onSubmitEditing: this.selectNextInput,
                      value: this.state.depth ? this.state.depth.toString() : "",
                    }}
                    style={{ margin: 10 }}
                  />
                  <View style={{ flex: 1, flexDirection: "row", alignItems: "center", margin: 10 }}>
                    <Label>Units</Label>
                    <Toggle selected={this.state.unit === "CM"} left="CM" right="IN" onPress={this.updateUnit} />
                  </View>
                </Row>
              </View>
            </ScrollView>
          </DoneButton>
        </ConsignmentBG>
        {
          //  When we want to show a picker, it should replace the keyboard, so move the
          //  keyboard down and push up the ConsignmentBG to fill in the space
        }
        {this.state.showPicker
          ? <Picker
              style={{ height: 220, backgroundColor: "black", marginTop: 40 }}
              key="picker"
              selectedValue={this.state.category}
              onValueChange={this.changeCategoryValue}
            >
              {categoryOptions.map(opt =>
                <Picker.Item color="white" label={opt.name} value={opt.value} key={opt.value} />
              )}
            </Picker>
          : null}
      </View>
    )
  }
}
