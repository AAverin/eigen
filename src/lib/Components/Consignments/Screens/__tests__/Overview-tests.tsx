import * as React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import Artist from "../Artist"
import Location from "../Location"
import Metadata from "../Metadata"
import Overview from "../Overview"

import Provenance from "../Provenance"
import SelectFromPhotoLibrary from "../SelectFromPhotoLibrary"
import Welcome from "../Welcome"

const nav = {} as any
const route = {} as any

it("Sets up the right view hierarchy", () => {
  const tree = renderer.create(<Overview navigator={nav} route={route} setup={{}} />).toJSON()
  expect(tree).toMatchSnapshot()
})

describe("Opening the right page", () => {
  let navigator = {} as any
  let overview: Overview

  beforeEach(() => {
    navigator = { push: jest.fn() } as any
    overview = new Overview({ navigator, route, setup: {} })
  })

  it("pushes a Location when you tap on Locations", () => {
    overview.goToLocationTapped()
    expect(navigator.push).toBeCalledWith({ component: Location })
  })

  it("pushes a Artist when you tap on Artist", () => {
    overview.goToArtistTapped()
    expect(navigator.push).toBeCalledWith({ component: Artist, passProps: expect.anything() })
  })

  it("pushes a Metadata when you tap on Metadata", () => {
    overview.goToMetadataTapped()
    expect(navigator.push).toBeCalledWith({ component: Metadata, passProps: expect.anything() })
  })

  it("pushes a Provenance when you tap on Provenance", () => {
    overview.goToProvenanceTapped()
    expect(navigator.push).toBeCalledWith({ component: Provenance, passProps: expect.anything() })
  })

  it("pushes a SelectFromPhotoLibrary when you tap on Photos", () => {
    overview.goToPhotosTapped()
    expect(navigator.push).toBeCalledWith({ component: SelectFromPhotoLibrary, passProps: expect.anything() })
  })
})
