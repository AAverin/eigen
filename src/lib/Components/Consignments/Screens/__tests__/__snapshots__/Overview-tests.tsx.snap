// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets up the right view hierarchy 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "black",
          "height": "100%",
          "maxWidth": 540,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "paddingTop": 18,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 30,
                  "marginBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "textAlign": "center",
                },
                Object {
                  "marginLeft": 40,
                  "marginRight": 40,
                },
                Object {
                  "fontFamily": "AGaramondPro-Regular",
                },
              ]
            }
          >
            Complete work details to submit
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#cccccc",
                  "fontSize": 20,
                  "paddingLeft": 25,
                  "paddingRight": 25,
                },
                Object {
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "AGaramondPro-Italic",
                },
              ]
            }
          >
            Provide as much detail as possible so that our partners can best assess your work.
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "black",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderColor": "white",
                        "borderRadius": 13,
                        "borderWidth": 1,
                        "height": 26,
                        "justifyContent": "center",
                        "marginRight": 20,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AvantGardeGothicITC",
                      },
                      undefined,
                    ]
                  }
                >
                  ARTIST/DESIGNER
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AGaramondPro-Regular",
                        "paddingTop": 6,
                        "textAlign": "right",
                      },
                      undefined,
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderColor": "white",
                        "borderRadius": 13,
                        "borderWidth": 1,
                        "height": 26,
                        "justifyContent": "center",
                        "marginRight": 20,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AvantGardeGothicITC",
                      },
                      undefined,
                    ]
                  }
                >
                  PHOTOS
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderColor": "white",
                        "borderRadius": 13,
                        "borderWidth": 1,
                        "height": 26,
                        "justifyContent": "center",
                        "marginRight": 20,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AvantGardeGothicITC",
                      },
                      undefined,
                    ]
                  }
                >
                  METADATA
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AGaramondPro-Regular",
                        "paddingTop": 6,
                        "textAlign": "right",
                      },
                      undefined,
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderColor": "white",
                        "borderRadius": 13,
                        "borderWidth": 1,
                        "height": 26,
                        "justifyContent": "center",
                        "marginRight": 20,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AvantGardeGothicITC",
                      },
                      undefined,
                    ]
                  }
                >
                  LOCATION
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AGaramondPro-Regular",
                        "paddingTop": 6,
                        "textAlign": "right",
                      },
                      undefined,
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  undefined,
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "height": 60,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderColor": "white",
                        "borderRadius": 13,
                        "borderWidth": 1,
                        "height": 26,
                        "justifyContent": "center",
                        "marginRight": 20,
                        "width": 26,
                      },
                      undefined,
                    ]
                  }
                >
                  <Image
                    source={1}
                  />
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AvantGardeGothicITC",
                      },
                      undefined,
                    ]
                  }
                >
                  PROVENANCE
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "fontFamily": "AGaramondPro-Regular",
                        "paddingTop": 6,
                        "textAlign": "right",
                      },
                      undefined,
                    ]
                  }
                >
                  
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#e5e5e5",
                    "height": 1,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Array [
                Object {
                  "justifyContent": "center",
                },
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingVertical": 6,
                },
              ]
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(51, 51, 51, 1)",
                  "borderColor": "rgba(51, 51, 51, 1)",
                  "borderWidth": 1,
                  "height": 43,
                  "justifyContent": "center",
                  "marginTop": 20,
                  "width": 174,
                }
              }
              testID={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontSize": 12,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 1)",
                      "fontFamily": "AGaramondPro-Regular",
                      "opacity": 1,
                    },
                    Object {
                      "fontFamily": "AvantGardeGothicITC",
                    },
                  ]
                }
              >
                NEXT
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "position": "absolute",
            "right": 20,
            "top": 20,
            "width": 40,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={1}
      />
    </View>
  </View>
</View>
`;
